### ISSUE TITLE
Allow mods to customize how light looks per player

### ISSUE THREAD

**Issue by rdococ** [minetest/minetest#8890]
It would be cool if mods were able to change how the lighting looks per player.

For example, if you enter some kind of hell dimension, you might want it to have some level of ambient light, so even the darkest light level would look a dim orange - or if you're wearing night vision goggles, all light levels would look bright.

This could be accomplished either by simply allowing mods to define four 'base colors' (for darkness, 100% artificial light only, 100% natural light only, and 100% both), or allowing mods to provide an array or image of what color should be used at every single possible light level.

This wouldn't provide support for true colored lighting, but it might help to give some atmosphere to different worlds or dimensions.

**Comment by ClobberXD** [comment]
I like this idea - it's a more generic, flexible alternative to full-bright mode.

**Comment by ZedekThePD** [comment]
Agreed, this could be very useful for different dimensions and even day/night atmosphere.

**Comment by benrob0329** [comment]
Couldn't you override the day/night ratio and use a semitransparent HUD overlay for most circumstances?

**Comment by LoneWolfHT** [comment]
HUD can be disabled with f1
